hq/app/controllers/SnykController.scala

```diff
@@ -0,0 +1,71 @@
+package controllers
+
+
+import auth.SecurityHQAuthActions
+import play.api.Configuration
+import play.api.libs.ws.WSClient
+import play.api.mvc._
+
+import scala.concurrent.ExecutionContext
+import logic.SnykDisplay
+import api.Snyk
+import com.gu.configraun.Errors.ConfigraunError
+import utils.attempt.{Attempt, Failure}
+import utils.attempt.PlayIntegration.attempt
+import model._
+
+class SnykController(val config: Configuration, val configraun: com.gu.configraun.models.Configuration)
+                    (implicit
+                     val ec: ExecutionContext,
+                     val wsClient: WSClient,
+                     val bodyParser: BodyParser[AnyContent],
+                     val controllerComponents: ControllerComponents,
+                     val assetsFinder: AssetsFinder)
+  extends BaseController  with SecurityHQAuthActions
+{
+
+  def snyk: Action[AnyContent] = {
+
+    Action.async {
+      attempt {
+        for {
+          token <- getSnykToken
+          requiredOrganisation <- getSnykOrganisation
+
+          organisationResponse <- Snyk.getSnykOrganisations(token, wsClient)
+          organisation <- SnykDisplay.getOrganisation(organisationResponse.body, requiredOrganisation)
+
+          projectResponse <- Snyk.getProjects(token, organisation.id, wsClient)
+          projects <- SnykDisplay.getProjectIdList(projectResponse.body)
+          labelledProjects = SnykDisplay.labelOrganisations(projects, organisation)
+
+          vulnerabilitiesResponse <- Snyk.getProjectVulnerabilities(organisation.id, projects, token, wsClient)
+          vulnerabilitiesResponseBodies = vulnerabilitiesResponse.map(a => a.body)
+          parsedVulnerabilitiesResponse <- SnykDisplay.parseProjectVulnerabilities(vulnerabilitiesResponseBodies)
+
+          results = SnykDisplay.labelProjects(labelledProjects, parsedVulnerabilitiesResponse)
+          sortedResult = SnykDisplay.sortProjects(results)
+        } yield Ok(views.html.snyk.snyk(sortedResult))
+      }
+    }
+
+  }
+
+  def getSnykToken: Attempt[SnykToken] = configraun.getAsString("/snyk/token") match {
```

Comment:
I mentioned on the last PR that these functions perhaps shouldn't be in the controller (they aren't actions). Do we need them at all? If this config is required why don't we fail at start time rather than at runtime whenever this is requested?

Reply:
I don't like moving them out of the controller into AppComponents, as I think AppComponents shouldn't need to know what configuration elements the SnykController needs.  However, I don't like runtime failures either.

Suggestions gratefully received.

Reply:
AppComponents is responsible for starting the application so by that measure it would need to be aware of what the application requires (and thus, what the SnykController needs).

Leave it as-is for now and we can explore this together when we're both in the office. I do think the config changes here will need a bit of thought either way, so fine to tackle this as part of that process.